pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/TableRebalanceAttemptContext.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.controller.helix.core.rebalance;
+
+import com.google.common.annotations.VisibleForTesting;
+
+
+/**
+ * Track the job configs and attempt numbers as part of the job ZK metadata for future retry.
+ */
+public class TableRebalanceAttemptContext {
```

Comment:
Suggest just naming it `TableRebalanceContext`. Retry is just one usage of this context, and we may add more usage in the future (e.g. `jobId` is not only used in retry)

Reply:
in fact, `Attempt` was meant to include both original and retry runs. but sgtm to name this more generic.